docs/integrations/fastapi.md

```diff
@@ -47,9 +47,15 @@ The `GraphQLRouter` accepts the following options:
   interface.
 - `allow_queries_via_get`: optional, defaults to `True`, whether to enable
   queries via `GET` requests
-- `context_getter`: optional FastAPI dependency for providing custom context
-  value.
-- `root_value_getter`: optional FastAPI dependency for providing custom root
+- `context_getter`: optional, default FastAPI dependency for providing custom
+  context value.
+- `http_get_context_getter`: optional, override `context_getter` for queries
+  executed via HTTP GET.
+- `http_post_context_getter`: optional, override `context_getter` for queries
+  executed via HTTP POST.
+- `ws_context_getter`: optional, override `context_getter` for queries executed
+  via WebSockets.
```

Comment:
wouldn't be nice to keep context getter and pass the request so one can make the decision of what to return based on that?

Reply:
So I'll start by saying there is a non-trivial chance I don't understand your suggestion.

What this PR is trying to do is allow you to inject a different set of dependencies based on the transport type.

In the current api `context_getter` is directly attached to the endpoint as a dependency. This means that the signature of that method defines what gets injected at runtime (basically it's a dag). If your `context_getter` doesn't depend on anything (or it's dependencies are common for all transports) than it's trivial to do an `if` statement on the request method and conditionally do something different. But if one of your dependencies does something specific for the transpor, than you can't use FastAPI to inject those dependencies because there is no way to tell the dependency injection system to only inject them for a given transport.

That said if we don't want to expand the API, the FastAPI integration is pretty simply and I'm happy to just fork it into my code.

Or if you have a more elegant way to expose this functionality I'm happy to update the PR.